### PR TITLE
Implement ReportDateIndicator

### PR DESCRIPTION
**Pull request by MrRefactor**
### Details

This PR concentrates on implementing Date dividers within chats to distinguish Today, Yesterday and other dates. We add an indicator between each day of the week that changes between messages, giving indication when that message was sent in an easy way. 
### Fixed Issues

$ #18101 
PROPOSAL: [#18101 (comment)](https://github.com/Expensify/App/issues/18101#issuecomment-1569941817)

### Tests

1. Open chat,
2. Check if the "moving" indicator is showing correct date,
3. Scroll the chat up and down between dates to check if "moving" indicator is correctly switching dates,
4. Verify that "static" indicators are showing correct dates 

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as tests

### QA Steps

Same as tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/32591413/311016c1-bc8a-4fac-9f0b-e42691b480e3

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/32591413/be6dc134-db47-4580-a455-ac324c9fb026

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/32591413/ab3a0db2-e760-4311-89fa-fc434fb5a781

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/32591413/8824f01f-b84c-448d-a3d9-cdb526d25f2d

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/32591413/71ef66bd-ddbb-4d7b-98f8-1b5824ba03e0

https://github.com/Expensify/App/assets/32591413/feae5e61-ef9e-4692-9edd-5595ab421771

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/32591413/4b29fefa-e22e-4b21-a226-e4f96a8fb0d1

</details>

### CODE REVIEW

File: src/libs/DateUtils.ts [hunk 1]
```diff
@@ -176,44 +176,58 @@ function isYesterday(date: Date, timeZone: SelectedTimezone): boolean {
...
-        return `${todayAt} ${format(date, CONST.DATE.LOCAL_TIME_FORMAT)}${tz}`;
+        return `${today}`;
     }
     if (isTomorrow(date, currentSelectedTimezone)) {
-        return `${tomorrowAt} ${format(date, CONST.DATE.LOCAL_TIME_FORMAT)}${tz}`;
+        return `${tomorrow}`;
     }
     if (isYesterday(date, currentSelectedTimezone)) {
-        return `${yesterdayAt} ${format(date, CONST.DATE.LOCAL_TIME_FORMAT)}${tz}`;
+        return `${yesterday}`;
```
Comment (Swor71) [inline src/libs/DateUtils.ts:210]: `today`, `tomorrow` and `yesterday` should already be strings, right?

File: src/pages/home/report/ReportActionsList.js [hunk 2]
```diff
@@ -179,6 +182,49 @@ function ReportActionsList({
...
+    );
+
+    const onViewableItemsChanged = useCallback(
+        ({viewableItems}) => {
+            if (viewableItems.length <= 0) {
+                return null;
+            }
+
+            const firstVisibleItem = viewableItems[viewableItems.length - 1];
+            const {index, isViewable} = firstVisibleItem;
```
Comment (Swor71) [inline src/pages/home/report/ReportActionsList.js:219]: nit: you can merge these two lines

**Review by rezkiy37 — COMMENTED** [review]
Left comments, after LGTM.

**Comment by shawnborton** [comment]
cc @Expensify/design for a review on this one

I wonder if we should add like 4px of vertical padding to the rows that contain the dates? This way we don't get the row:hover effect from chat messages bumping right up against it like this:
CleanShot 2024-02-28 at 08 45 19@2x

Notice that Slack has something similar going on: 
CleanShot 2024-02-28 at 08 45 59@2x

**Comment by shawnborton** [comment]
And maybe in doing that, the top-anchored badge will be just slightly further away from the top header bar - I think a little extra padding/gap would be nice:
CleanShot 2024-02-28 at 08 46 43@2x

**Comment by shawnborton** [comment]
Do you have an example to show with the new message indicator (green button)?

**Comment by MrRefactor** [comment]
> Do you have an example to show with the new message indicator (green button)?

here you go!


https://github.com/Expensify/App/assets/32591413/824f2802-a4d3-4c9b-b0f6-618957743fdd

**Comment by dannymcclain** [comment]
The new message indicator is looking good!

@shawnborton I agree the date badges need some space. In my mocks I have `8px` of top and bottom padding around them. From the video @MrRefactor posted, it looks like the new message indicator also sits 8px away from the top.

CleanShot 2024-02-28 at 07 59 10

**Comment by MrRefactor** [comment]
> The new message indicator is looking good!
> 
> @shawnborton I agree the date badges need some space. In my mocks I have `8px` of top and bottom padding around them. From the video @MrRefactor posted, it looks like the new message indicator also sits 8px away from the top.
> 

My bad, I will apply the changes soon!

**Comment by shawnborton** [comment]
Wonderful, do you have new screenshots for review?

**Comment by MrRefactor** [comment]
> Also, going to run a quick test build so we can test as well.

Screenshot 2024-02-28 at 22 42 26
Screenshot 2024-02-28 at 22 42 19

**Comment by shawnborton** [comment]
Going to leave some comments now that I am testing!

For this case:
CleanShot 2024-02-28 at 10 18 24


I would think that the Feb 6 date badge would be on top of the Feb 2 badge as we scroll. I also think the Feb 2 badge should stop scrolling with the message a bit sooner - it doesn't need to keep scrolling with the page even if the Feb 2 message is aleady gone. Let me know if that makes sense.

**Comment by shawnborton** [comment]
Found a case of the same date being duplicated:
CleanShot 2024-02-28 at 10 20 19@2x

**Comment by shawnborton** [comment]
I noticed that the badges have a pointer cursor on them, yet interacting with them does nothing. Let's remove the pointer since tapping them doesn't actually do anything.

**Comment by shawnborton** [comment]
@Expensify/design @puneetlath @JmillsExpensify @trjExpensify do you think we should keep the badge behavior even in things like expense threads? 
CleanShot 2024-02-28 at 10 22 09@2x

I guess the answer is yes for consistency... though it does add a bit to that view.

**Comment by shawnborton** [comment]
Also in this case, why does one date have the year and the other doesn't? 
CleanShot 2024-02-28 at 10 23 08@2x

Maybe we can drop the year if you are looking at a date from the current year. So we would never show 2024 right now, just 2023 and before.

**Comment by MrRefactor** [comment]
> Also in this case, why does one date have the year and the other doesn't? CleanShot 2024-02-28 at 10 23 08@2x
> 
> Maybe we can drop the year if you are looking at a date from the current year. So we would never show 2024 right now, just 2023 and before.

I will check and try to address all of the comments tomorrow!

**Comment by dannymcclain** [comment]
@shawnborton re: [your comment](https://github.com/Expensify/App/pull/35897#issuecomment-1969212207)
>I would think that the Feb 6 date badge would be on top of the Feb 2 badge as we scroll. I also think the Feb 2 badge should stop scrolling with the message a bit sooner - it doesn't need to keep scrolling with the page even if the Feb 2 message is aleady gone. Let me know if that makes sense.

I think it would be ideal if the badged "pushed" the other out of the way, like how Slack does this:

CleanShot 2024-02-28 at 09 38 52

>Maybe we can drop the year if you are looking at a date from the current year. So we would never show 2024 right now, just 2023 and before.

Agree!

>do you think we should keep the badge behavior even in things like expense threads?
> I guess the answer is yes for consistency... though it does add a bit to that view.

Oof. I don't really like it... But I see what you mean about consistency. It _is_ just another room/thread after all...

**Comment by shawnborton** [comment]
> I think it would be ideal if the badged "pushed" the other out of the way, like how Slack does this:

Oh yes, I love that!

**Comment by MrRefactor** [comment]
> Found a case of the same date being duplicated:

fixed

> I noticed that the badges have a pointer cursor on them, yet interacting with them does nothing. Let's remove the pointer since tapping them doesn't actually do anything.

fixed

> Maybe we can drop the year if you are looking at a date from the current year. So we would never show 2024 right now, just 2023 and before.

fixed

**Comment by MrRefactor** [comment]
> Going to leave some comments now that I am testing!
> 
> For this case: CleanShot 2024-02-28 at 10 18 24 [ CleanShot 2024-02-28 at 10 18 24 ](https://private-user-images.githubusercontent.com/2319350/308603402-7adc0b69-2ed7-4d41-b614-299f89acd883.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkyMTM4MTEsIm5iZiI6MTcwOTIxMzUxMSwicGF0aCI6Ii8yMzE5MzUwLzMwODYwMzQwMi03YWRjMGI2OS0yZWQ3LTRkNDEtYjYxNC0yOTlmODlhY2Q4ODMuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjlUMTMzMTUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGZlMWVmNmYyNmRlYTRiODMwZjNlZGExZGI5NjNkYzc2MmZiNTcxMzdjZWYwODdiNzFlM2ZlYmEyZGQ2ZmQ0MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.71gdg-PrAVzvVHCQouIF1Mg5KsxAGSQEexCmbzbB5Cg) [ ](https://private-user-images.githubusercontent.com/2319350/308603402-7adc0b69-2ed7-4d41-b614-299f89acd883.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkyMTM4MTEsIm5iZiI6MTcwOTIxMzUxMSwicGF0aCI6Ii8yMzE5MzUwLzMwODYwMzQwMi03YWRjMGI2OS0yZWQ3LTRkNDEtYjYxNC0yOTlmODlhY2Q4ODMuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjlUMTMzMTUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGZlMWVmNmYyNmRlYTRiODMwZjNlZGExZGI5NjNkYzc2MmZiNTcxMzdjZWYwODdiNzFlM2ZlYmEyZGQ2ZmQ0MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.71gdg-PrAVzvVHCQouIF1Mg5KsxAGSQEexCmbzbB5Cg)
> 
> I would think that the Feb 6 date badge would be on top of the Feb 2 badge as we scroll. I also think the Feb 2 badge should stop scrolling with the message a bit sooner - it doesn't need to keep scrolling with the page even if the Feb 2 message is aleady gone. Let me know if that makes sense.

In that case we have to keep "Feb 2" badge on top, in other cases it would be covered by the content of the messages when scrolling

**Comment by MrRefactor** [comment]
> @shawnborton re: [your comment](https://github.com/Expensify/App/pull/35897#issuecomment-1969212207)
> 
> > I would think that the Feb 6 date badge would be on top of the Feb 2 badge as we scroll. I also think the Feb 2 badge should stop scrolling with the message a bit sooner - it doesn't need to keep scrolling with the page even if the Feb 2 message is aleady gone. Let me know if that makes sense.
> 
> I think it would be ideal if the badged "pushed" the other out of the way, like how Slack does this:
> 
> CleanShot 2024-02-28 at 09 38 52 
> 

Implementation of that behaviour requires big refactor in the codebase, I would suggest to stay with the current state for now a create a separated issue for that 

cc: @dannymcclain

**Comment by dannymcclain** [comment]
>Implementation of that behaviour requires big refactor in the codebase, I would suggest to stay with the current state for now a create a separated issue for that

Ok that makes sense. I still think we need to figure out a way to handle/improve [this](https://github.com/Expensify/App/pull/35897#issuecomment-1969212207). It feels really weird with the older date staying over top of the newer one. @shawnborton any other ideas?

**Comment by puneetlath** [comment]
>Implementation of that behaviour requires big refactor in the codebase, I would suggest to stay with the current state for now a create a separated issue for that

Just so I understand, why is this?

**Comment by MrRefactor** [comment]
> > Implementation of that behaviour requires big refactor in the codebase, I would suggest to stay with the current state for now a create a separated issue for that
> 
> Just so I understand, why is this?

As for now we are using absolutely positioned date indicator component to change date values as we scroll, "static" date indicator components that are showing days between messages are part of inverted flat list that we are using. So in order to make "push" behaviour possible we would need to incorporate that "moving" indicator into flat list and then work based on that.

**Comment by chiragsalian** [comment]
odd, i was waiting for C+ review but i just noticed they are not assigned. Not sure why I'm assigned instead of puneet. Assigning puneet so this is on his radar since it's his issue. Should we assign C+ to review as well @puneetlath?

**Comment by MrRefactor** [comment]
Do we have any update on this? @shawnborton @puneetlath

**Comment by puneetlath** [comment]
Ok, so I've been thinking about it more and I don't think we should merge this without being able to have the current date push out the past date. We can definitely do a feature branch if you think you'd prefer to split that up into separate PRs @MrRefactor. 

Or alternatively we can have the date be in a fixed position in the chat and not hover over messages. And only implement the hover once we are able to have the current date push the last date out. 

Thoughts on those options?

**Comment by dannymcclain** [comment]
@puneetlath, `#not-an-engineer` but either of those sound better than merging as is to me.

**Comment by MrRefactor** [comment]
> 
> Or alternatively we can have the date be in a fixed position in the chat and not hover over messages. And only implement the hover once we are able to have the current date push the last date out.
> 
This one sounds good to me, as I understand we should delete "flowing" indicator and leave static ones between messages, right?

**Comment by puneetlath** [comment]
Yes, exactly. And then in a follow-up PR we can add the "flowing" indicator, but have it properly replace the dates. Sound good?

**Comment by MrRefactor** [comment]
@MrMuzyk Will work on v1

**Comment by MrRefactor** [comment]
hey @chiragsalian @puneetlath PR is ready for review, @MrMuzyk fixed minor issues and v1 is ready!

**Comment by puneetlath** [comment]
In general, it seems to look as-designed to me. @Expensify/design what do you think? 

image

**Comment by shawnborton** [comment]
For example, this thread is from March 7 but you have no idea once you are in the thread itself:
CleanShot 2024-04-29 at 08 36 58@2x

**Comment by dannymcclain** [comment]
>Visually I think it looks okay. I do miss the floating indicators though.

Yeah the badge style looks good, but dang floating would be better. I'm kinda having trouble knowing what date certain messages were posted on without the date in the timestamp. Anyone else? Like if a certain date has a _bunch_ of messages, you can't see the date indicator and the message at the same time, so it's hard to know the actual date of the message. 

>I just tested this in our #social room and I'm not seeing any indicator for "Today" - is that a bug? I only see "Yesterday" as the most recent date indicator.

I'm also not seeing a `Today` indicator in social, but I can see it in my personal DM. And sometimes I'm seeing `Yesterday` (or another date) repeated multiple times.

**Comment by shawnborton** [comment]
> Anyone else?

Yup, I feel the same!

**Comment by MrMuzyk** [comment]
>I'm also not seeing a Today indicator in social, but I can see it in my personal DM. And sometimes I'm seeing Yesterday (or another date) repeated multiple times.

I will look into this


>I do miss the floating indicators though.

This is something that will be done in a followup PR in the future, is that correct? From what I've understood from handover with @MrRefactor this is a `v1` that we want to get merged in the current state

**Comment by shawnborton** [comment]
> This is something that will be done in a followup PR in the future, is that correct? 

I think that was the idea but after seeing it in action, it makes me wonder if we should try to implement it correctly the first time, otherwise we're going to be in a weird state without the floating labels for a while. Do we have any idea how long a follow up would take, and what the technical feasibility is like? Just want to make sure we have a solid plan and that we aren't just kicking this can down the road.

**Comment by shawnborton** [comment]
Also, I think we really need to sort out our approach for threads and reports. Do you have thoughts there @puneetlath? 

For example, it doesn't feel super great here:
CleanShot 2024-04-30 at 08 31 57@2x

Not to mention that report should actually display as a single transaction view, so maybe you need to pull in main?

**Comment by MrMuzyk** [comment]
>Do we have any idea how long a follow up would take, and what the technical feasibility is like?

I will investigate this and I will try to get back to you with an answer today but no promises as I will be finishing for the day soon (I will be back 6th of May)

>Not to mention that report should actually display as a single transaction view, so maybe you need to pull in main?

Recording was done after merging main today in the morning and I'm pretty sure changes from this branch didn't add/change anything else but the indicator. I will rebase again

**Comment by shawnborton** [comment]
Sounds good, will trigger a new test build

**Comment by MrMuzyk** [comment]
[This was](https://github.com/Expensify/App/pull/35897#issuecomment-1972365960) @MrRefactor idea for the implementation of the floating indicator. I will discuss this with him, maybe search for some kind of alternative as he mentioned costly refactor and get back with an estimate

**Comment by shawnborton** [comment]
Seeing some weirdness with timezones maybe?

This PR:
CleanShot 2024-04-30 at 10 57 48@2x

Staging:
CleanShot 2024-04-30 at 10 58 19@2x

**Comment by shawnborton** [comment]
Labels don't show up in the one-expense view of a report:
CleanShot 2024-04-30 at 10 59 08@2x

**Comment by puneetlath** [comment]
So am I right in understanding: 
1. We still think the ReportDateIndicator is better than how we currently do dates in the timestamp
2. We want to use it everywhere -- threads, transactions, etc
3. We only want to use it if we can do the "flowing" indicator, as the fixed position one is not good enough 

Is that right? If so, maybe we assess the technical complexity of doing the flowing indicator. And based on that we decide whether we want to go forward or just punt this for now.

**Comment by shawnborton** [comment]
That sounds about right, though I am a bit torn on this point:

>We want to use it everywhere -- threads, transactions, etc

It can definitely make the threads a little messy, especially if you are nested a few levels deep across a range of dates/times in a thread. What do you think we should do there?

**Comment by puneetlath** [comment]
I agree with you that it looks a little weird. But I also don't love the idea of using different styles in different rooms. Which is why I'm kind of questioning the feature altogether 😅

**Comment by shawnborton** [comment]
Haha yeah, I think we'd want to see these floating labels in those places to better make our decision. @MrMuzyk can you see if you can get the pill labels to show up in places like threads as well?

**Comment by dannymcclain** [comment]
>We still think the ReportDateIndicator is better than how we currently do dates in the timestamp

Yeah right now I'm not totally sold that this is better than how we're currently doing it.

**Comment by MrMuzyk** [comment]
>Seeing some weirdness with timezones maybe?

I can look into this

---
>Labels don't show up in the one-expense view of a report:

>can you see if you can get the pill labels to show up in places like threads as well?

This is weird because they are showing up for me when I'm running this locally. Screenshots below

Screenshot 2024-05-07 at 12 23 21

Screenshot 2024-05-07 at 12 23 36

image

---
>I'm still not seeing a "Today" label in the #social room either.

Screenshot 2024-05-07 at 12 26 28

Same case as above. It is working fine for me 🤔 

---

@puneetlath @shawnborton 

When it comes to implementing floating indicators I've had a chat with @MrRefactor today and we both do not see the way of implementing indicators in the same manner as slack does in the current state of the app. The part where new indicator pushes the old one is the most complex bit that we don't know how to tackle. We both think that having opinion from someone who is more experienced with animations might be helpful - @WoLewicki

**Comment by WoLewicki** [comment]
Hey, is there a reason why I have been mentioned here?

**Comment by MrMuzyk** [comment]
@WoLewicki Hey, @MrRefactor was working on a feature introducing date indicators and now I'm helping him with this as he's busy with RN upgrade. Some time ago there was a [sugestion](https://github.com/Expensify/App/pull/35897#issuecomment-1969263577) to create this in the same manner as slack is handling this. We have a `v1` that is different but we want to estimate and plan how much work would it be to make it work and look like in the slack example. As I've mentioned in the previous comment we've had a chat today with @MrRefactor regarding this and we couldn't see a clear plan on how to achieve the same effect as in slack app. He suggested that it would be great to get your opinion on this feature. Do you think it is achievable in the current state of the app?

**Comment by WoLewicki** [comment]
Hey! I am no expert in this manner unfortunately 😅 I guess you could probably implement it as sticky headers? I can see there is an issue in `FlashList` repo regarding having it in inverted mode: https://github.com/Shopify/flash-list/issues/611. Did you investigate this approach at all?

Another approach with the current implementation would be to pass information about the current scroll offset and mimic the behavior of sticky headers the core (https://github.com/facebook/react-native/blob/cc99e924ca4f8d29cb6bf6bb5d2e180df541936a/packages/react-native/Libraries/Components/ScrollView/ScrollViewStickyHeader.js). 

> Do you think it is achievable in the current state of the app?

I think there are no hard limitations that could prevent us from doing it.

**Comment by MrMuzyk** [comment]
Thanks for the input 👍 

So the option with sticky headers was investigated but we can't go this path because of 2 reasons:
1. Like you've mentioned the inverted sticky header doesn't work properly at the moment
2. We're not even sure if it would do the trick in our case. It could work when scrolling down for sure but we're not 100% convinced if it would work as well when scrolling up. We can't even test it now because inverted sticky header isn't working 😞 We could prepare some kind of mvp on regular list and see if it would work.

Custom solution might be the way to go if we want to implement it in the near future.

It would also be great to have full requirements on how the date dividers should behave in multiple scenarios. I know that we have this gif showcasing how it works on slack but having a bit more than that would also help us decide on the possible approach in the future.

@puneetlath @shawnborton 
Can you check if the current version still misbehaves for you? If so, can you please provide some videos/screenshots and steps to reproduce your issues? In [this comment](https://github.com/Expensify/App/pull/35897#issuecomment-2098052218) I've provided some screenshots showcasing that it works fine for me in multiple scenarios. On top of that I've fixed the issue with timezones that was pointed out :)

**Comment by puneetlath** [comment]
My take is that given the complexity of trying to make the floating headers, we should just set this project aside for now. We have other more pressing concerns (e.g. performance) that I think we're better of focusing on. What do y'all think?

**Comment by shawnborton** [comment]
Yup, that makes sense and I agree with that.

**Comment by puneetlath** [comment]
Ok thanks for your effort on this everyone! Closing out for now.

**Pull request closed without merging**